english-helper/src/context/DarkModeContext.tsx

```diff
@@ -0,0 +1,47 @@
+import { createContext, type ReactNode, useContext, useEffect } from "react";
+import { useLocalStorage } from "../hooks/useLocalStorage";
+
+type DarkModeState = {
+  isDarkMode: boolean;
+};
+type DarkModeContextValue = DarkModeState & {
+  toggleDarkMode: () => void;
+};
+
+type DarkModeProviderProps = {
+  children: ReactNode;
+};
+
+const DarkModeContext = createContext<DarkModeContextValue | null>(null);
+
+function DarkModeProvider({ children }: DarkModeProviderProps) {
+  const [isDarkMode, setIsDarkMode] = useLocalStorage(false, "isDarkMode");
+
+  useEffect(() => {
+    const html = document.documentElement;
+    if (isDarkMode) {
+      html.classList.add("dark");
+    } else {
+      html.classList.remove("dark");
+    }
+  });
```

Comment:
}, [isDarkMode]);

I think here we need to add dependency array to avoid unnecesary render